File: app/assets/javascripts/profile/ConnectService.js
(function () {
    var app = angular.module('shocktrade');

    /**
     * Connect Service
     * @author lawrence.daniels@gmail.com
     */
    app.factory('ConnectService', function ($http, $q) {
        var service = {};

        service.deleteMessages = function (messageIDs) {
            var deferred = $q.defer();
            $http({
                method: 'DELETE',
                url: '/api/updates',
                data: angular.toJson(messageIDs)
            }).success(function (data, status, headers, config) {
                deferred.resolve(data[0]);
            }).error(function (response) {
                deferred.reject(response);
            });
            return deferred.promise;
        };

        service.getUserInfo = function (fbUserId) {
            var deferred = $q.defer();
            $http.get('/api/profile/facebook/' + fbUserId)
                .success(function (data, status, headers, config) {
                    deferred.resolve(data[0]);
                })
                .error(function (data, status, headers, config) {
                    deferred.reject("Error: " + symbol + "-" + data + "(" + status + ")");
                });
            return deferred.promise;
        };

        service.getUserUpdates = function (userName, limit) {
            return $http.get('/api/updates/' + userName + '/' + limit);
        };

        service.identifyFacebookFriends = function (fbFriends) {
            // build the JSON query list
            var list = [];
            for (var n = 0; n < fbFriends.length; n++) {
                list.push(fbFriends[n].id);
            }

            // make the service call
            var deferred = $q.defer();
            $http({
                method: 'POST',
                url: '/api/profile/facebook/friends',
                data: angular.toJson(fbFriends)
            }).success(function (data, status, headers, config) {
                deferred.resolve(data);
            }).error(function (data, status, headers, config) {
                deferred.reject("Error: " + symbol + "-" + data + "(" + status + ")");
            });
            return deferred.promise;
        };

        return service;
    });

})();

File: app-js/src/main/scala/com/shocktrade/javascript/profile/ConnectService.scala
package com.shocktrade.javascript.profile

import biz.enef.angulate.Service
import biz.enef.angulate.core.HttpService

import scala.scalajs.js
import scala.scalajs.js.annotation.JSExportAll

/**
 * Connect Service
 * @author lawrence.daniels@gmail.com
 */
@JSExportAll
class ConnectService($http: HttpService) extends Service {

  def deleteMessages: js.Function = (messageIDs: js.Array[String]) => deleteMessages_@(messageIDs)

  def deleteMessages_@(messageIDs: js.Array[String]) = $http.delete[js.Dynamic]("/api/updates", messageIDs)

  def getUserInfo: js.Function = (facebookID: String) => getUserInfo_@(facebookID)

  def getUserInfo_@(facebookID: String) = $http.get[js.Dynamic](s"/api/profile/facebook/$facebookID")

  def getUserUpdates: js.Function = (userName: String, limit: Int) => getUserUpdates_@(userName, limit)

  def getUserUpdates_@(userName: String, limit: Int) = $http.get[js.Array[js.Dynamic]](s"/api/updates/$userName/$limit")

  def identifyFacebookFriends: js.Function = (fbFriends: js.Array[js.Dynamic]) => identifyFacebookFriends_@(fbFriends)

  def identifyFacebookFriends_@(fbFriends: js.Array[js.Dynamic]) = $http.post[js.Dynamic]("/api/profile/facebook/friends", fbFriends.map(_.id))


}
